zebra-chain/src/transaction.rs

```diff
@@ -146,4 +154,33 @@ impl Transaction {
                 Some(TransparentInput::Coinbase { .. })
             )
     }
+
+    // TODO(jlusby): refine type
+    pub fn sighash(&self, network: Network, height: BlockHeight, hash_type: u32) -> Hash {
+        sighash::SigHasher {
+            trans: self,
+            network,
+            height,
+            hash_type,
+        }
+        .sighash()
+    }
+
+    fn header(&self) -> u32 {
+        match self {
+            Transaction::V1 { .. } => 1,
+            Transaction::V2 { .. } => 2,
+            Transaction::V3 { .. } => 3 | 1 << 31,
+            Transaction::V4 { .. } => 4 | 1 << 31,
+        }
+    }
+
+    fn group_id(&self) -> Option<u32> {
+        match self {
+            Transaction::V1 { .. } => None,
+            Transaction::V2 { .. } => None,
+            Transaction::V3 { .. } => Some(OVERWINTER_VERSION_GROUP_ID),
+            Transaction::V4 { .. } => Some(SAPLING_VERSION_GROUP_ID),
+        }
+    }
```

Comment:
I think defining these functions could be avoided if we don't try to share common code between serialization and sighash implementations: https://github.com/ZcashFoundation/zebra/pull/870/files#r471234178